src/controllers/ProductsController.ts

```diff
@@ -0,0 +1,296 @@
+import { Request, Response, NextFunction } from 'express';
+import ProductService from '../ProductService';
+import { Product as PrismaProduct, Prisma, Like } from '@prisma/client';
+import prisma from '../index';
+import { ProductCreateDto, ProductUpdateDto } from '../dtos/ProductDto';
+import CommentService from '../CommentService';
+import LikeService from '../LikeService';
+import CommentRepository from '../repositories/CommentRepository';
+import LikeRepository from '../repositories/LikeRepository';
+
+class ProductsController {
+  private productService: ProductService;
+  private commentService: CommentService;
+  private likeService: LikeService;
+
+  constructor(productService: ProductService) {
+    this.productService = productService;
+    const commentRepository = new CommentRepository();
+    this.commentService = new CommentService(commentRepository);
+    const likeRepository = new LikeRepository();
+    this.likeService = new LikeService(likeRepository);
+  }
```

Comment:
해당 ProductsController를 여러 route에서 사용하게 된다면 여러 생성자들이 추가로 생성하게 될 수 있어요. 의존성 주입 일관성을 위해서 라우터 조립부에서 생성해서 컨트롤러 생성자에 주입하는 것이 좋습니다.
```suggestion
  constructor(productService: ProductService, commendService: CommentService, likeService: LikeService) {
    this.productService = productService;
    this.commentRepository = commentService
    this.likeRepository = likeRepository
  }
```